File: 2356-largest-combination-with-bitwise-and-greater-than-zero/2356-largest-combination-with-bitwise-and-greater-than-zero.cpp
class Solution {
public:
    int largestCombination(vector<int>& candidates) {
        
        int size = candidates.size();

        vector<int>res(25,0);
        bitset<25>b[size];

        for(int i=0;i<size;i++){
            b[i] = candidates[i];
        }
        int mxLenght = 0;

        for(int i = 0 ;i < size ;i++){
            
            for(int j = 24 ; j >= 0;j--){
                res[j] += b[i][j];
                mxLenght = max(mxLenght , res[j]);
            }
        }
        return mxLenght;
    }
};

File: 2356-largest-combination-with-bitwise-and-greater-than-zero/2356-largest-combination-with-bitwise-and-greater-than-zero.cs
public class Solution {
    public int LargestCombination(int[] candidates) {
        int n = candidates.Length;
        //int mxLength = 0 ;
        int[] res = new int[25];
        for(int i = 24; i>= 0 ;i--){
            foreach(int u in candidates){
                if((u & ( 1 << i))!=0) res[i]++;
            }
            
        }
        return res.Max();
    }
}